src/main.ts

```diff
@@ -147,19 +147,20 @@ async function run(): Promise<void> {
 
     const issueKeys = Jira.getJIRAIssueKeys(headBranch).concat(Jira.getJIRAIssueKeys(title));
     if (!issueKeys.length) {
-      const body = Jira.getNoIdComment(headBranch);
+      const body = Jira.getNoIdComment();
       const comment = { ...commonPayload, body };
       await gh.upsertComment('no-id', comment);
 
-      return exit('JIRA issue id is missing in your branch.');
+      return exit('JIRA issue id is missing in your branch or PR title.');
     }
 
     // use the last match (end of the branch name)
     const issueKey = issueKeys[issueKeys.length - 1];
     console.log(`JIRA key -> ${issueKey}`);
 
-    const details: JIRADetails = await jira.getTicketDetails(issueKey);
-    if (details.key) {
+    try {
```

Comment:
Why are we allowing getTicketDetails to throw an exception, We can make it return null if there is no data and keep the if condition as is

Reply:
I tried this, but that meant amending the JIRADetails interface to be nullable, which made the assignment break in hilarious ways, et.c, et.c
So yes, this is an issue of skipping abstraction layers, but I must admit my TypeScript isn't good enough to do it in the correct way. Very happy for code suggestions of how to do it correctly. :)

Reply:
Sure you are welcome; you will just need to allow undefined as a type.
  const details: JIRADetails | undefined = await jira.getTicketDetails(issueKey);

I know TS not look like other languages but by time you will like it :D